packages/p2p/src/components/my-ads/my-ads-row-renderer.jsx

```diff
@@ -35,9 +35,14 @@ const MyAdsRowRenderer = observer(({ row: advert, setAdvert }) => {
     const amount_dealt = amount - remaining_amount;
     const is_buy_advert = type === buy_sell.BUY;
 
-    const onClickActivateDeactivate = () => {
-        my_ads_store.onClickActivateDeactivate(id, is_advert_active, setIsAdvertActive);
+    const onClickAddPaymentMethod = () => {
+        if (!general_store.is_barred) {
+            setAdvert(advert);
+            my_ads_store.showQuickAddModal(advert);
+        }
     };
+    const onClickActivateDeactivate = () =>
+        !general_store.is_barred && my_ads_store.onClickActivateDeactivate(id, is_advert_active, setIsAdvertActive);
```

Comment:
There's already an `onClickActivateDeactivate` function in my ads store so why don't we just check for `general_store.is_barred` there?

Reply:
Replaced condition check from component to store, thx for notice!